AppManager/app_manager_server.py

```diff
@@ -631,6 +631,12 @@ def remove_conflicting_jars(app_name):
     "/appengine-jsr107cache-*.jar", shell=True)
   subprocess.call("rm -f " + lib_dir + \
     "/jsr107cache-*.jar", shell=True)
+  subprocess.call("rm -f " + lib_dir + \
+    "/appengine-mapreduce*.jar", shell=True)
+  subprocess.call("rm -f " + lib_dir + \
+    "/appengine-pipeline*.jar", shell=True)
+  subprocess.call("rm -f " + lib_dir + \
```

Comment:
I think it's best to avoid making shell calls when possible. In this case, you can use Python's native methods for filesystem operations. You can use `os.listdir` and then check each file against a list of patterns (`['appengine-api-stubs-*.jar', 'appengine...']`).